### PR TITLE
Exposes union type item connection; deprecates existing

### PR DESCRIPTION
**Pull request by dzucconi**
```gql
{
  feature(id: "alserkal-art-week") {
    name
    setsConnection(first: 10) {
      totalCount
      edges {
        node {
          __typename
          id
          name
          orderedItemsConnection(first: 10) {
            totalCount
            edges {
              node {
                ... on FeaturedLink {
                  title
                  image {
                    url
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

```json
{
  "data": {
    "feature": {
      "name": "Alserkal Art Week",
      "setsConnection": {
        "totalCount": 2,
        "edges": [
          {
            "node": {
              "__typename": "OrderedSet",
              "id": "T3JkZXJlZFNldDo1ZTZmNWVlMmM1MmE1OTAwMTExMTEyMWY=",
              "name": "Latest News",
              "orderedItemsConnection": {
                "totalCount": 1,
                "edges": [
                  {
                    "node": {
                      "title": "Upcoming Exhibitions at Alserkal Avenue",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/_SzbXBMwj9JIj0QFspYcZQ/large_rectangle.jpg"
                      }
                    }
                  }
                ]
              }
            }
          },
          {
            "node": {
              "__typename": "OrderedSet",
              "id": "T3JkZXJlZFNldDo1ZTZhNGE5NDFjYjdjYjAwMTJmMGNlNTY=",
              "name": "Exhibitions",
              "orderedItemsConnection": {
                "totalCount": 30,
                "edges": [
                  {
                    "node": {
                      "title": "1x1 Art Gallery",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/SFWBRqOlZV7dN0JImc9AAg/wide.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Ayyam Gallery",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/aKK02kNS3xVUR1fSAgJsFQ/large_square.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Carbon 12",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/bpzb9jNuKsRrqw-nZVH_qA/wide.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Concrete",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/u8b2hFUhNo1CHg2gfgbU5A/small_rectangle.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Elmarsa",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/i9ZS_hh3NtS1a1Af0RFB8w/small_square.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Gallery Isabelle van den Eynde",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/Q46GkTuAZVguljpbK0GhaQ/wide.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Green Art Gallery",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/4ePpPzVtk_QOfimFP4bcRA/medium_rectangle.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Grey Noise",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/0UGlX1CttmayUiPOYXQWYg/medium_rectangle.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Gulf Photo Plus",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/VeZMpqL1-OapRItdyXoJmQ/wide.jpg"
                      }
                    }
                  },
                  {
                    "node": {
                      "title": "Ishara Art Foundation",
                      "image": {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/b_FKZNNJ3fm3ZPg1Hoh_sw/medium_rectangle.jpg"
                      }
                    }
                  }
                ]
              }
            }
          }
        ]
      }
    }
  },
  "extensions": {
    "requests": {
      "gravity": {
        "requests": {
          "feature/alserkal-art-week": {
            "time": "0s 1.334ms",
            "cache": true,
            "length": "N/A"
          },
          "sets?owner_id=alserkal-art-week&owner_type=Feature&page=1&size=10&total_count=true": {
            "time": "0s 1.18ms",
            "cache": true,
            "length": "N/A"
          },
          "set/5e6f5ee2c52a59001111121f/items?page=1&size=10&total_count=true": {
            "time": "0s 2.229ms",
            "cache": true,
            "length": "N/A"
          },
          "set/5e6a4a941cb7cb0012f0ce56/items?page=1&size=10&total_count=true": {
            "time": "0s 3.455ms",
            "cache": true,
            "length": "N/A"
          }
        }
      }
    },
    "requestID": "e940bc20-9397-11ea-9dc0-df68a4d59d58"
  }
}
```

### CODE REVIEW

File: _schemaV2.graphql [hunk 1]
```diff
@@ -5922,6 +5929,26 @@ type OrderedSetEdge {
 
 union OrderedSetItem = Artist | Artwork | FeaturedLink | Gene
 
+# A connection to a list of items.
+type OrderedSetItemConnection {
+  # Information to aid in pagination.
+  pageInfo: PageInfo!
+
+  # A list of edges.
+  edges: [OrderedSetItemEdge]
```
Comment (dzucconi) [inline _schemaV2.graphql:5938]: Realistically all the Relay connections should be non-nullable, yeah? I think we don't do this anywhere but it should be safe and avoid all of the inevitable edges?...node?.id stuff that's going to happen when we move to strict null checks.

Comment (sweir27) [reply _schemaV2.graphql:5938]: There's a bigger discussion on that here: https://github.com/artsy/metaphysics/pull/2346 !

File: src/schema/v2/ordered_set.ts [hunk 2]
```diff
@@ -38,13 +40,40 @@ export const OrderedSetType = new GraphQLObjectType<
...
           return items.map(item => {
-            item.item_type = item_type // eslint-disable-line no-param-reassign
-            return item
+            return { ...item, item_type }
           })
         })
       },
     },
+
+    nextItemsConnection: {
```
[the remark this answers is not in the document]
Comment (dzucconi) [reply src/schema/v2/ordered_set.ts:49]: Good call. I was going to just remove the old one, then move this into its place.

But actually these are called OrderedItems in Gravity so either `setItemsConnection` or `orderedItemsConnection` is better and would let me avoid the deprecation dance if we're fine with that.

[the remark this answers is not in the document]
Comment (sweir27) [reply src/schema/v2/ordered_set.ts:49]: Oooh I like `orderedItemsConnection`.

File: _schemaV2.graphql (hunk 1 above)
Comment (dzucconi) [reply _schemaV2.graphql:5938]: Ah, exciting PR, that.

File: src/schema/v2/ordered_set.ts [hunk 3]
```diff
@@ -38,13 +40,40 @@ export const OrderedSetType = new GraphQLObjectType<
...
           return items.map(item => {
-            item.item_type = item_type // eslint-disable-line no-param-reassign
-            return item
+            return { ...item, item_type }
           })
         })
       },
     },
+
+    orderedItemsConnection: {
```
Comment (dzucconi) [inline src/schema/v2/ordered_set.ts:49]: Renamed!

**Pull request merged**